### ISSUE TITLE
Question: should bundled entry exports work? If so, how?

### ISSUE THREAD

**Issue by birdspider** [laravel-mix/laravel-mix#3200]
-   Laravel Mix Version: 6.0.41
-   Node Version: v17.3.0
-   NPM Version: 8.3.1
-   OS: Linux

### Description:

What steps does one need to take, to access `entry` exports in js in the browser? 
In my linked example repo `app.js` exports stuff, and also pollutes `window`. 

In the `public/index.html` I can (of course) access the window property, but I'm unable to `import X from '/bundled/app.js`  when using  laravel-mix. 

(If I async `import()` it I do get something which prints as: `Object { Symbol(Symbol.toStringTag): "Module" }`)
​
If I do vanilla modules, that is, I include `unbundled/app.js` then the import resolves correctly. 

So in short, why does:

* test-case `1) import X from '/bundled/app.js` not work,  [source1](https://github.com/birdspider/laravel-mix-example/blob/main/public/index.html#L34)
* when `4) import X from '/unbundled/app.js` does work? [source4](https://github.com/birdspider/laravel-mix-example/blob/main/public/index.html#L58)

What am I missing ?

I'm not sure if this is a:

* missunderstanding on my part or
* a documentation issue (i.e. feature not supported) or
* or an actual laravel mix (webpack) issue

### Steps To Reproduce:

checkout tiny example: https://github.com/birdspider/laravel-mix-example

**Comment by birdspider** [comment]
@noogen thx, while your reply alone did not change anything, thinking about it helped me to figure out that it is probably webpack related:

this works for me, but I have to test its limits:

```
// seems to work in dev and prod on FF, Chrome
mix.webpackConfig({
  experiments: {
    outputModule: true,
  },
  output: {
    module: true,
    library: {
      type: 'module',
    },
  }
});
```
